frontend/packages/knative-plugin/src/components/add/event-sources/KafkaSourceSection.tsx

```diff
@@ -1,37 +1,103 @@
-  const { values } = useFormikContext<FormikValues>();
-  const {
-    data: {
-      kafkasource: { bootstrapServers, topics },
-    },
-  } = values;
+  const [bootstrapServers, setBootstrapServers] = React.useState<SelectInputOption[]>([]);
+  const [bsPlaceholder, setBsPlaceholder] = React.useState<React.ReactNode>('');
+  const [kafkaTopics, setKafkaTopics] = React.useState<SelectInputOption[]>([]);
+  const [ktPlaceholder, setKtPlaceholder] = React.useState<React.ReactNode>('');
+
+  const memoResources = React.useMemo(() => strimziResourcesWatcher(), []);
+  const { kafkas, kafkatopics } = useK8sWatchResources<{
+    [key: string]: K8sResourceKind[];
+  }>(memoResources);
+
+  React.useEffect(() => {
+    let bootstrapServersOptions: SelectInputOption[] = [];
+    let placeholder: React.ReactNode = '';
+    if (kafkas.loaded && !kafkas.loadError) {
+      bootstrapServersOptions = !_.isEmpty(kafkas.data)
+        ? _.map(getBootstrapServers(kafkas.data), (bs) => ({
+            value: bs,
+            disabled: false,
+          }))
+        : [
+            {
+              value: 'No Bootstrapservers found',
+              disabled: true,
+            },
+          ];
+      placeholder = 'Add Bootstrapservers';
+    } else if (kafkas.loadError) {
+      bootstrapServersOptions = [{ value: kafkas.loadError?.message, disabled: true }];
+      placeholder = 'Error loading Bootstrapservers';
+    } else {
+      bootstrapServersOptions = [{ value: 'Loading Bootstrapservers...', disabled: true }];
+      placeholder = '...';
+    }
+    setBootstrapServers(bootstrapServersOptions);
+    setBsPlaceholder(placeholder);
+  }, [kafkas.data, kafkas.loaded, kafkas.loadError]);
+
+  React.useEffect(() => {
+    let topicsOptions: SelectInputOption[] = [];
+    let placeholder: React.ReactNode = '';
+    if (kafkatopics.loaded && !kafkatopics.loadError) {
+      topicsOptions = !_.isEmpty(kafkatopics.data)
+        ? _.map(kafkatopics.data, (kt) => ({
+            value: kt?.metadata.name,
+            disabled: false,
+          }))
+        : [
+            {
+              value: 'No Topics found',
+              disabled: true,
+            },
+          ];
+      placeholder = 'Add Topics';
+    } else if (kafkatopics.loadError) {
+      topicsOptions = [{ value: kafkatopics.loadError?.message, disabled: true }];
+      placeholder = 'Error loading Topics';
+    } else {
+      topicsOptions = [{ value: 'Loading Topics...', disabled: true }];
+      placeholder = '...';
+    }
+    setKafkaTopics(topicsOptions);
+    setKtPlaceholder(placeholder);
+  }, [kafkatopics.data, kafkatopics.loaded, kafkatopics.loadError]);
+
   return (
     <FormSection title="KafkaSource" extraMargin>
-      <TextColumnField
+      <SelectInputField
         data-test-id="kafkasource-bootstrapservers-field"
         name="data.kafkasource.bootstrapServers"
-        label="BootstrapServers"
-        addLabel="Add Bootstrapservers"
+        label="Bootstrapservers"
```

Comment:
Bad find replace? There are several instances where the two words are joined as one: `bootstrapservers`
```suggestion
        label="Bootstrap Servers"
```

Reply:
No. Just wanted to keep it same as it was before as it was UX approved. I can change that for all KafkaSource fields then.

Reply:
yeah i see issue in two fields `BootstrapServers` and `ConsumerGroup `, we should update as suggested to `Bootstrap Servers` and `Consumer Group` cc @bgliwa01

Reply:
Updated at all places.